web/secondary_toolbar.js

```diff
@@ -29,173 +29,228 @@
+   * @param {EventBus} eventBus
+   */
+  function SecondaryToolbar(options, eventBus) {
     this.toolbar = options.toolbar;
-    this.buttonContainer = this.toolbar.firstElementChild;
-
-    // Define the toolbar buttons.
     this.toggleButton = options.toggleButton;
     this.presentationModeButton = options.presentationModeButton;
-    this.openFile = options.openFile;
-    this.print = options.print;
-    this.download = options.download;
-    this.viewBookmark = options.viewBookmark;
-    this.firstPage = options.firstPage;
-    this.lastPage = options.lastPage;
-    this.pageRotateCw = options.pageRotateCw;
-    this.pageRotateCcw = options.pageRotateCcw;
-    this.toggleHandTool = options.toggleHandTool;
+    this.openFileButton = options.openFile;
+    this.printButton = options.print;
+    this.downloadButton = options.download;
+    this.viewBookmarkButton = options.viewBookmark;
+    this.firstPageButton = options.firstPage;
+    this.lastPageButton = options.lastPage;
+    this.pageRotateCwButton = options.pageRotateCw;
+    this.pageRotateCcwButton = options.pageRotateCcw;
+    this.toggleHandToolButton = options.toggleHandTool;
     this.documentPropertiesButton = options.documentPropertiesButton;
 
-    // Attach the event listeners.
-    var elements = [
-      // Button to toggle the visibility of the secondary toolbar:
-      { element: this.toggleButton, handler: this.toggle },
-      // All items within the secondary toolbar
-      { element: this.presentationModeButton,
-        handler: this.presentationModeClick },
-      { element: this.openFile, handler: this.openFileClick },
-      { element: this.print, handler: this.printClick },
-      { element: this.download, handler: this.downloadClick },
-      { element: this.viewBookmark, handler: this.viewBookmarkClick },
-      { element: this.firstPage, handler: this.firstPageClick },
-      { element: this.lastPage, handler: this.lastPageClick },
-      { element: this.pageRotateCw, handler: this.pageRotateCwClick },
-      { element: this.pageRotateCcw, handler: this.pageRotateCcwClick },
-      { element: this.toggleHandTool, handler: this.toggleHandToolClick },
-      { element: this.documentPropertiesButton,
-        handler: this.documentPropertiesClick }
-    ];
-
-    for (var item in elements) {
-      var element = elements[item].element;
-      if (element) {
+    this.eventBus = eventBus;
+
+    this.opened = false;
+    this.previousContainerHeight = null;
+    this.newContainerHeight = null;
+    this.buttonContainer = this.toolbar.firstElementChild;
+
+    // Bind the event listeners for click and hand tool actions.
+    this._bindClickListeners();
+    this._bindHandToolListener();
+  }
+
+  SecondaryToolbar.prototype = {
+    /**
+     * @return {boolean}
+     */
+    get isOpen() {
+      return this.opened;
+    },
+
+    _bindClickListeners: function SecondaryToolbar_bindClickListeners() {
+      var elements = [
```

Comment:
Since this PR is a good opportunity improve the code, I've been wondering if we couldn't simplify it even further.
Given that we no longer access various functions in the `SecondaryToolbar` from event handlers in `app.js`, I'm not sure if having all the functions below (e.g. `presentationModeButtonClick`) is still necessary.
@timvandermeij What do you think about something along these lines, which would mean that most of the helper methods below is no longer necessary?

```javascript
  var self = this;

  // Button to toggle the visibility of the secondary toolbar.
  this.toggleButton.addEventListener('click', self.toggle);

  // All items within the secondary toolbar.
  var buttons = [
    { element: this.presentationModeButton, eventName: 'presentationmode', shouldClose: true },
    .
    .
    .
    { element: this.pageRotateCwButton, eventName: 'rotatecw', shouldClose: false },
    .
    .
    .
  ];

  for (var button in buttons) {
    var element = buttons[button].element;
    var eventName = buttons[button].eventName:
    var shouldClose = buttons[button].shouldClose;

    element.addEventListener('click', function (evt) {
      self.eventBus.dispatch(eventName);
      if (shouldClose) {
        self.close();
      }
    });
  }
```